nbgrader/apps/releaseapp.py

```diff
@@ -0,0 +1,97 @@
+import os
+import sys
+import shutil
+
+from IPython.utils.traitlets import Unicode, List, Bool
+
+from nbgrader.apps.baseapp import TransferApp, transfer_aliases, transfer_flags
+from nbgrader.utils import self_owned
+
+
+aliases = {}
+aliases.update(transfer_aliases)
+aliases.update({
+})
+
+flags = {}
+flags.update(transfer_flags)
+flags.update({
+    'force': (
+        {'ReleaseApp' : {'force' : True}},
+        "Force overwrite of existing files in the exchange."
+    ),
+    'remove': (
+        {'ReleaseApp' : {'remove': True}},
+        "Only remove existing files in the exchange."
+    ),
+})
+
+class ReleaseApp(TransferApp):
+
+    name = u'nbgrader-release'
+    description = u'Release an assignment to the nbgrader exchange'
+
+    aliases = aliases
+    flags = flags
+
+    examples = """
+        Here we go...
+        """
+
+    force = Bool(False, config=True, help="Force overwrite existing files in the exchange.")
+
+    remove = Bool(False, config=True, help="Only remove existing files in the exchange.")
+
+    def init_args(self):
+        if len(self.extra_args) == 1:
+            self.assignment_id = self.extra_args[0]
+        else:
+            self.fail("Invalid number of argument, call as `nbgrader release ASSIGNMENT`.")
+
+    def init_src(self):
+        self.src_path = os.path.abspath(os.path.join(self.release_directory, self.assignment_id))
+        if not os.path.isdir(self.src_path):
+            self.log.error("Assignment not found: {}/{}".format(self.release_directory, self.assignment_id))
+            self.fail("You have to run `nbgrader release` from your main nbgrader directory.")
```

Comment:
Is it possible this error could be generated if they hadn't run `nbgrader assign` first? Could we maybe check whether the source directory exists, and if it does, tell them the they need to run `nbgrader assign` first; otherwise, give them this error about running from the main nbgrader directory.